cilium-cli/connectivity/tests/egressgateway.go

```diff
@@ -110,14 +162,17 @@ func (s *egressGateway) Run(ctx context.Context, t *check.Test) {
 	i := 0
 	for _, client := range ct.ClientPods() {
 		for _, dst := range ct.HostNetNSPodsByNode() {
-			t.NewAction(s, fmt.Sprintf("ping-%d", i), &client, &dst, features.IPFamilyV4).Run(func(a *check.Action) {
-				a.ExecInPod(ctx, ct.PingCommand(dst, features.IPFamilyV4))
+			t.ForEachIPFamily(func(ipFam features.IPFamily) {
+				t.NewAction(s, fmt.Sprintf("ping-%s-%d", ipFam, i), &client, &dst, ipFam).Run(func(a *check.Action) {
+					a.ExecInPod(ctx, ct.PingCommand(dst, ipFam))
+				})
 			})
 			i++
 		}
 	}
 
 	// DNS query (pod to service connectivity). Should not get masqueraded with egress IP
+	// This test case fails for ipv6, might need to make changes for this to work with kube-dns or target a different in-cluster service.
```

Comment:
Highlighting for reviewers, as this is the only test-case in the suite that I couldn't port to ipv6.

Reply:
Do we need to set up a cluster using a CoreDNS service with an IPv6 ClusterIP to pass this test case? There was a similar issue in the node-local-dns test as well.
https://github.com/cilium/cilium/issues/38444

Reply:
Yes, I think that would make this work. But since that test doesn't test anything specific to EGW, we probably don't need to block the PR on that.